Build failure: forgejo-lts

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

build forgejo

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310376466

### Relevant log output

```console
--- FAIL: TestParseGitURLs (0.00s)
    --- FAIL: TestParseGitURLs/https://[fe80:14fc:cec5:c174:d88%2510]:20/go-gitea/gitea.git (0.00s)
        url_test.go:162:
                Error Trace:    /build/source/modules/git/url/url_test.go:162
                Error:          Received unexpected error:
                                parse "https://[fe80:14fc:cec5:c174:d88%2510]:20/go-gitea/gitea.git": invalid host: ParseAddr("fe80:14fc:cec5:c174:d88%10"): address string too short
                Test:           TestParseGitURLs/https://[fe80:14fc:cec5:c174:d88%2510]:20/go-gitea/gitea.git
FAIL
FAIL    forgejo.org/modules/git/url     0.012s
FAIL
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.52, NixOS, 25.11 (Xantusia), 25.11.20251012.cf3f5c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/la80d03vcf555bkcjp84cij5c4i63m5n-source`

### Notify maintainers


@emilylange @bendlas @adamcstephens @pyrox0 @tebriel 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fo/forgejo/generic.nix
+++ b/pkgs/by-name/fo/forgejo/generic.nix
@@ -12,6 +12,7 @@
   bash,
   brotli,
   buildGoModule,
+  fetchpatch,
   forgejo,
   git,
   gzip,
@@ -83,6 +84,14 @@ buildGoModule rec {
 
   patches = [
     ./static-root-path.patch
+  ]
+  ++ lib.optionals lts [
+    (fetchpatch {
+      # fix for go 1.25.2 stricter ipv6 parsing, remove for LTS > 11.0.6
+      name = "fix-test-ipv6-go125.patch";
+      url = "https://codeberg.org/forgejo/forgejo/commit/0d9a8e3fa2cf9228290ed1a9a5767e6ba204edd7.patch";
+      hash = "sha256-AM4/kgCXSU5Bj8aOObm6qyeL1SEpeFhmlT42lMJ2o08=";
+    })
   ];
 
   postPatch = ''

